cmd/pinsharesvc/process.go

```diff
@@ -0,0 +1,501 @@
+	logPath := filepath.Join(pm.config.DataDirectory, "logs", "ipfs.log")
+	logFile, err := os.OpenFile(logPath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to open IPFS log file: %w", err)
+	}
+	pm.ipfsLogFile = logFile
+
+	// Create command
+	pm.ipfsCmd = exec.CommandContext(ctx, pm.config.IPFSBinary, "daemon")
+	pm.ipfsCmd.Env = append(os.Environ(),
+		fmt.Sprintf("IPFS_PATH=%s", ipfsDataPath),
+	)
+	pm.ipfsCmd.Stdout = logFile
+	pm.ipfsCmd.Stderr = logFile
+
+	// Set process group for proper cleanup on Windows
+	pm.ipfsCmd.SysProcAttr = &syscall.SysProcAttr{
+		CreationFlags: syscall.CREATE_NEW_PROCESS_GROUP,
+	}
+
+	// Start the process
+	if err := pm.ipfsCmd.Start(); err != nil {
+		logFile.Close()
+		return fmt.Errorf("failed to start IPFS daemon: %w", err)
+	}
+
+	pm.logInfo(fmt.Sprintf("IPFS daemon started with PID %d", pm.ipfsCmd.Process.Pid))
+
+	// Create exit channel and monitor process in background
+	pm.ipfsExited = make(chan struct{})
+	go pm.monitorProcess(ctx, pm.ipfsCmd, "IPFS", pm.ipfsExited)
+
+	return nil
+}
+
+// initializeIPFS initializes IPFS in the data directory
+func (pm *ProcessManager) initializeIPFS() error {
+	ipfsDataPath := pm.config.GetIPFSDataPath()
+
+	cmd := exec.Command(pm.config.IPFSBinary, "init")
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("IPFS_PATH=%s", ipfsDataPath),
+	)
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("ipfs init failed: %w\nOutput: %s", err, string(output))
+	}
+
+	pm.logInfo("IPFS initialized successfully")
+
+	// Configure IPFS settings
+	if err := pm.configureIPFS(); err != nil {
+		return fmt.Errorf("failed to configure IPFS: %w", err)
+	}
+
+	return nil
+}
+
+// configureIPFS configures IPFS settings from PinShare config.json
+// This must be called when IPFS is NOT running (no repo.lock)
+func (pm *ProcessManager) configureIPFS() error {
+	ipfsDataPath := pm.config.GetIPFSDataPath()
+	env := append(os.Environ(), fmt.Sprintf("IPFS_PATH=%s", ipfsDataPath))
+
+	pm.logInfo(fmt.Sprintf("Configuring IPFS ports: API=%d, Gateway=%d, Swarm=%d",
+		pm.config.IPFSAPIPort, pm.config.IPFSGatewayPort, pm.config.IPFSSwarmPort))
+
+	// Set API port
+	if err := pm.runIPFSConfig(env, "Addresses.API", fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", pm.config.IPFSAPIPort)); err != nil {
+		return err
+	}
+
+	// Set Gateway port
+	if err := pm.runIPFSConfig(env, "Addresses.Gateway", fmt.Sprintf("/ip4/127.0.0.1/tcp/%d", pm.config.IPFSGatewayPort)); err != nil {
+		return err
+	}
+
+	// Set Swarm port - include all transport protocols (TCP, UDP/QUIC, WebRTC, WebTransport)
+	// Must use --json flag since this is a JSON array
+	port := pm.config.IPFSSwarmPort
+	swarmAddrs := fmt.Sprintf(`["/ip4/0.0.0.0/tcp/%d", "/ip6/::/tcp/%d", "/ip4/0.0.0.0/udp/%d/webrtc-direct", "/ip4/0.0.0.0/udp/%d/quic-v1", "/ip4/0.0.0.0/udp/%d/quic-v1/webtransport", "/ip6/::/udp/%d/webrtc-direct", "/ip6/::/udp/%d/quic-v1", "/ip6/::/udp/%d/quic-v1/webtransport"]`,
```

Comment:
Let's open a comprehensive issue regarding adding support via the settings dialog for enumerating, and selecting specific (or all) interfaces and IP versions instead of ALWAYS binding to all interfaces on all versions. Then let's add a comment here which references this new issue: `// TODO_IMPROVE(#XXX): add support for binding specif and wildcard interfaces and/or IP versions`.